pandas/core/indexes/datetimes.py

```diff
@@ -2245,17 +2245,47 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
     --------
     pandas.period_range : Return a fixed frequency PeriodIndex.
     pandas.interval_range : Return a fixed frequency IntervalIndex.
+    Numpy daterange: `numpy daterange<https://docs.scipy.org/doc/numpy/reference/arrays.datetime.html>`__.
 
     Examples
     --------
-    >>> pd.date_range('2018-10-03', periods=2) # doctest: +NORMALIZE_WHITESPACE
-    DatetimeIndex(['2018-10-03', '2018-10-04'], dtype='datetime64[ns]',
-                   freq='D')
-
-    >>> pd.date_range(start='2018-01-01', end='20180103')
-    ... # doctest: +NORMALIZE_WHITESPACE
-    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03'],
-                    dtype='datetime64[ns]', freq='D')
+
+    Given the `start` and `end` mandatory parameters,
+    with default ``freq='D'`` (Days)
+
+    >>> pd.date_range(start='1/1/2018', end='1/10/2018')
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+                   '2018-01-05', '2018-01-06', '2018-01-07', '2018-01-08',
+                   '2018-01-09', '2018-01-10'],
+                  dtype='datetime64[ns]', freq='D')
+
+    Given the ``start`` and ``periods`` mandatory parameters,
+    periods would be the limit of the DatetimeIndex, with default `freq='D'`
+    (D=calendar day frequency)
+
+    >>> pd.date_range(start='1/1/2018', periods=5)
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+                   '2018-01-05'],
+                  dtype='datetime64[ns]', freq='D')
+
+    Changed the `freq` (frequency) to `'M'` (month end frequency) and `tz` to `'UTC'`
+
+    >>> pd.date_range(start='1/1/2018', periods=5, freq='M', tz='UTC')
+    DatetimeIndex(['2018-01-31', '2018-02-28', '2018-03-31', '2018-04-30',
+                   '2018-05-31'],
+                  dtype='datetime64[ns, UTC]', freq='M')
+
+    closed set to `'left'`, this makes the month range with the starting month
+    (inclusive of dec to feb)
+
+    >>> pd.date_range(start='31/12/2017', end='3/1/2018', freq='M', closed='left')
+    DatetimeIndex(['2017-12-31', '2018-01-31', '2018-02-28'], dtype='datetime64[ns]', freq='M')
+
+    closed set to `'right'`, this makes the month range after the starting month
+    (inclusive of jan to feb)
+
+    >>> pd.date_range(start='31/12/2017', end='3/1/2018', freq='M', closed='right')
+    DatetimeIndex(['2018-01-31', '2018-02-28'], dtype='datetime64[ns]', freq='M')
```

Comment:
Could you note that `MS` can be used for month start and show that?

Reply:
okay I am updating one more example with `MS`.